pyexchange/uniswapv2_analytics.py

```diff
@@ -388,53 +410,38 @@ def get_all_trades(self, pair: str, page_number: int = 1) -> List[Trade]:
-        start_time = two_days_ago_unix if two_days_ago_unix > self.all_last_pair_hour_timestamp else self.all_last_pair_hour_timestamp
+        # calculate starting block, assuming there's 15 seconds in a given block
+        current_block = self.get_current_block()
+        two_day_ago_block = int(current_block - (4 * 60 * 25))
 
-        get_pair_hour_datas_query = """query ($pair: Bytes!, $hourStartUnix: Int!)
-        {
-            pairHourDatas(where: {pair: $pair, hourStartUnix_gt: $hourStartUnix}) {
-                pair {
-                    id
-                    totalSupply
-                    token0 {
-                        id
-                    }
-                    token1 {
-                        id
-                    }
-                    token0Price
-                    token1Price
-                }
-                id
-                hourStartUnix
-                hourlyVolumeToken0
-                hourlyVolumeToken1
-                reserve0
-                reserve1
-            }
-        }
-        """
-        variables = {
-            'pair': pair_address.address.lower(),
-            'hourStartUnix': start_time
-        }
+        start_block = two_day_ago_block if two_day_ago_block > self.all_last_pair_block else self.all_last_pair_block
+        end_block = current_block
+
+        raw_block_trades = []
+        index = 0
+        checked_block = start_block
+        while checked_block < end_block:
+            block_trade = self.get_block_trade(pair_address, checked_block)[0]
+
+            raw_block_trades.append(block_trade)
+
+            if checked_block != start_block:
+                if raw_block_trades[index]['token0']['id'] == base_token.address.address:
+                    previous_base_token_reserves = Wad.from_number(raw_block_trades[index - 1]['reserve0'])
+                else:
+                    previous_base_token_reserves = Wad.from_number(raw_block_trades[index - 1]['reserve1'])
 
-        result = self.graph_client.query_request(get_pair_hour_datas_query, variables)
-        pair_hour_data_list = result['pairHourDatas']
+                timestamp = self.web3.eth.getBlock(checked_block).timestamp
 
-        for index, trade in enumerate(pair_hour_data_list):
-            if trade['pair']['token0']['id'] == base_token.address.address:
-                previous_base_token_reserves = Wad.from_number(pair_hour_data_list[0]['reserve0'])
-            else:
-                previous_base_token_reserves = Wad.from_number(pair_hour_data_list[index - 1]['reserve0'])
+                trades_list.append(UniswapTrade.from_all_trades_message(block_trade, pair, base_token, previous_base_token_reserves, timestamp))
 
-            formatted_trade = UniswapTrade.from_all_trades_message(trade, pair, base_token, previous_base_token_reserves)
-            trades_list.append(formatted_trade)
+            # assume there's 240 blocks in an hour
+            checked_block += 240
```

Comment:
On average there's 266.7 blocks per hour.  So we're checking 54 minutes of blocks here, which seems fine as an arbitrary way to batch the requests.

Reply:
Yeah i'm not super enthused with the blocks/hour assumption, but I was afraid of making a separate set of calls for each period to get the associated timestamp - block relationship.